controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_controller:
      type: forward_command_controller/ForwardCommandController

arm_controller:
  ros__parameters:
    joints:
      - gen3_joint_1
      - gen3_joint_2
      - gen3_joint_3
      - gen3_joint_4
      - gen3_joint_5
      - gen3_joint_6
      - gen3_joint_7
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
      gen3_joint_1: {trajectory: 0.01, goal: 0.01}
      gen3_joint_2: {trajectory: 0.01, goal: 0.01}
      gen3_joint_3: {trajectory: 0.01, goal: 0.01}
      gen3_joint_4: {trajectory: 0.01, goal: 0.01}
      gen3_joint_5: {trajectory: 0.01, goal: 0.01}
      gen3_joint_6: {trajectory: 0.01, goal: 0.01}
      gen3_joint_7: {trajectory: 0.01, goal: 0.01}

gripper_controller:
  ros__parameters:
    joints:
      - gen3_robotiq_85_left_knuckle_joint
    interface_name: position 